From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/85a408a07bcc
From: xen-devel@lists.xenproject.org
Patch-mainline: Never, SUSE-Xen specific
Subject: pciback: only check PF if actually dealing with a VF
References: bsc#999245

c/s c00c8be428cd ("pciback: check PF instead of VF for
PCI_COMMAND_MEMORY") has a blatant backporting mistake: ->is_physfn
gets set only for SR-IOV devices once they've got enabled; we really
need to check ->is_virtfn here.

Signed-off-by: Jan Beulich <jbeulich@suse.com>

--- a/drivers/xen/pciback/conf_space_capability_msi.c
+++ b/drivers/xen/pciback/conf_space_capability_msi.c
@@ -62,7 +62,7 @@ int pciback_enable_msix(struct pciback_d
 	 * But VF devices are unique in which the PF needs to be checked.
 	 */
 #ifdef CONFIG_PCI_IOV
-	phys_dev = dev->is_physfn ? dev : dev->physfn;
+	phys_dev = dev->is_virtfn ? dev->physfn : dev;
 #endif
 	pci_read_config_word(phys_dev, PCI_COMMAND, &cmd);
 	if (dev->msi_enabled || !(cmd & PCI_COMMAND_MEMORY))
